apps/wallet/src/ui/app/hooks/useRecentTransactions.ts

```diff
@@ -224,3 +221,30 @@ export function useRecentTransactions() {
         }
     );
 }
+
+export function useRecentTransaction(transactionDigest: string | null) {
+    const rpc = useRpc();
+    const userAddress = useAppSelector((state) => state.account.address);
+    return useQuery({
+        queryKey: [
+            'recent transaction',
+            'get transaction with effects',
+            transactionDigest,
+        ],
+        queryFn: async () => {
+            const transactionResponse = await rpc.getTransactionWithEffects(
+                transactionDigest!
+            );
+            const processedTransactions = await processTransactionEffects(
+                [transactionResponse],
+                userAddress!,
+                rpc
+            );
+            return processedTransactions?.[0] || null;
+        },
+        enabled: !!transactionDigest?.length && !!userAddress,
+        staleTime: 10 * 1000,
```

Comment:
This stale time can probably just be default, I just overrode it for the other one because I want activity view to feel fresh.